app/gui2/src/components/ToggleIcon.vue

```diff
@@ -0,0 +1,20 @@
+<script setup lang="ts">
+import SvgIcon from '@/components/SvgIcon.vue'
+import { ref } from 'vue'
+
+const props = defineProps<{ icon_name: String; modelValue?: Boolean }>()
```

Comment:
The prop names should use `camelCase`. Though here I would also keep it short, just `icon` is fine.